PixaView/Controller/PixaViewController.swift

```diff
@@ -0,0 +1,276 @@
+//
+//  PixaViewController.swift
+//  PixaView
+//
+//  Created by Дмитрий Игнатьев on 27.11.2021.
+//
+
+import UIKit
+import TestFramework
+
+class PixaViewController: UIViewController{
+    
+    //MARK: Outlets -
+    @IBOutlet weak var collectionView: UICollectionView!
+    
+    
+    //MARK:  Properties -
+    var hitsRESULT: [Hits] = [] //array from json
+    var enumValue = ""
+    var order: Order = .popular
+    var category: [Categories] = [.computer]
+    var currentPage = 1
+    var total = 0
+    var collectionViewFlowLayout = UICollectionViewFlowLayout()
+    //MARK: - end of Properties
+    
+    
+    
+    //MARK:  viewDidLoad -
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let sortButton = navigationItem.rightBarButtonItem = UIBarButtonItem(image: UIImage(systemName: "line.3.horizontal.circle"), style: .plain, target: self, action: #selector(pixaSortButton))
+        setupCollectionView()
+        fetch(order: order, currentPage: currentPage, filterCategory: category)
+    }
+    
+    //MARK: - view will layout
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        if UIDevice.current.orientation.isLandscape {
+            landscapeLayoutCollectionView()
+        }else if UIDevice.current.orientation.isPortrait{
+            portaitCollectionView()
+        }
+    }
+    
+    
+    // pixa sort button
+    @objc private func pixaSortButton(){
+        dismiss(animated: true, completion: nil)
```

Comment:
Usually we dissmiss/present when all related work are done, pls move it to the end of the method.